app/controllers/api/alert_definitions_controller.rb

```diff
@@ -1,26 +1,31 @@
 module Api
   class AlertDefinitionsController < BaseController
-    REQUIRED_FIELDS = %w(description db expression options).freeze
 
     before_action :set_additional_attributes
 
     def create_resource(type, id, data = {})
       assert_id_not_specified(data, type)
-      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
       begin
-        data["expression"] = MiqExpression.new(data["expression"])
+        create_miq_expression(data)
         data["enabled"] = true if data["enabled"].nil?
-        super(type, id, data).serializable_hash.merge("expression" => data["expression"])
+        super(type, id, data.deep_symbolize_keys).serializable_hash.merge("expression" => expression_values(data).first)
       rescue => err
         raise BadRequestError, "Failed to create a new alert definition - #{err}"
       end
     end
 
     def edit_resource(type, id = nil, data = {})
       raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
+      if expression_values(data).size == 1
```

Comment:
Is this still needed? Again, this is another business rule that needs to be encapsulated in the model.

Reply:
I will remove that too.
The intention here was to handle the case of replacing a miq expression with hash expression (and vice versa) when editing an alert, but we can leave that to the user to explicitly nullify these fields. 
cc @moolitayer @elad661